src/workload.py

```diff
@@ -59,7 +59,7 @@ def write(self, content: str, path: str) -> None:
 
     @override
     def exec(self, command: list[str], working_dir: str | None = None) -> str:
-        return str(self.container.exec(command, working_dir=working_dir).wait_output())
+        return self.container.exec(command, working_dir=working_dir).wait_output()[0]
```

Comment:
`wait_output` returns a tuple `(stdout, stderr)`. The previous version was just serialising it as a str, which was unconvenient for the `json.loads` part. Since on VM we do not care about stderr most of the time (we redirect it to `subprocess.PIPE` without acting on it) I figured it would be ok to do so here.

What I would prefer to do, though, is to use `httpx.get` because we would get a much better dev experience than `json.loads`-ing a curl command. I did not feel comfortable adding a big dependency out of nowhere for this ticket

Reply:
makes sense! We should try to provide the same interface on VM and K8s (although that they may have sometimes two different implementation). Using HTTPS call definitely go in that direction (also possibly having the same implementation)!

Anyhow...Up to you whether worth having it here or in a separate ticket, but I would definitely nudge to the HTTP which seems general to me

Reply:
Ah it's a tuple, all good then 👍🏾